solutions/swb-app/src/projectRoutes.ts

```diff
@@ -59,4 +67,45 @@ export function setUpProjectRoutes(router: Router, projectService: ProjectServic
       res.send(await projectService.createProject(request, res.locals.user));
     })
   );
+
+  // Soft delete project
+  router.put(
+    '/projects/:projectId/softDelete',
+    wrapAsync(async (req: Request, res: Response) => {
+      async function checkDependencies(projectId: string): Promise<void> {
+        // environments
+        const projectHasEnvironments = environmentService.doesDependencyHaveEnvironments(projectId);
+        // datasets
+        const projectHasDatasets = projectService.checkDependency('dataset', projectId);
```

Comment:
For `checkDependency`  we can use `metadataService.listDependentMetadata` that Sergey implemented [here](https://github.com/aws-solutions/solution-spark-on-aws/pull/641/files#diff-283a50193b6974f9ed0ce809bf1d2c13381291c2033491f2247b25ee999d1086)

Reply:
Sergey's `metadataService` is not in develop yet. I may merge as is for now and refactor before I merge feature to develop. Our methods are basically identical.